### ISSUE TITLE
New rule: selector-pseudo-class-no-unspaced-operator

### ISSUE THREAD

**Issue by fpuga** [stylelint/stylelint#4127]
> What is the problem you're trying to solve?

I'm reviewing _stylelint_ and _prettier_ to standardize a css formatting tool in our team. We are going to use _stylelint_ but prefer an output as much closer to _prettier_ as possible in many aspects.

One of the things that _prettier_ does is introduce a whitespace around operators in the pseudo class `nth-child`

As example, prettier converts this css:

```css
.table-checks td:nth-child(n+2) {
    text-align: center;
}
```
to this:

```css
.table-checks td:nth-child(n + 2) {
    text-align: center;
}
```

but there is not (or i don't find) an option to check this in _stylelint_

As there is already [a rule like this for functions](https://stylelint.io/user-guide/rules/function-calc-no-unspaced-operator/) it would make sense have another one for pseudo selectors.

> What solution would you like to see?

A new rule that implements the proposed check.

**Comment by evilebottnawi** [comment]
Sound good, maybe better name like selector-pseudo-class-an-plus-b-spaces with always and never value?

**Comment by hudochenkov** [comment]
Yeah, this rule could be in the core, I think. What name to choose is the hard question. Because it should support different cases.

This rule should support all of “simple cases”:

```css
a:nth-child(2n + 1) {}
a:nth-last-child(2n + 1) {}
a:nth-last-of-type(2n + 1) {}
a:nth-of-type(2n + 1) {}
```

And not break the following (not add spaces around `-`):

```css
a:not(:nth-child(-n + 3)) {}
```

And work with pseudo-classes inside pseudo-classes:

```css
a:not(:nth-child(2n + 1)) {}
```

And work with new syntax like [`:has()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:has#Examples):

```css
h1:has(+ p) {}
```

**Comment by evilebottnawi** [comment]
I think we should handle `h1:has(+ p) {}` and `h1:has(> p) {}` in https://stylelint.io/user-guide/rules/selector-combinator-space-after/ and https://stylelint.io/user-guide/rules/selector-combinator-space-before/, `selector-pseudo-class-an-plus-b-spaces` should handle only `an+b` grammar

**Comment by hudochenkov** [comment]
True. The only problem, no one knows what is `an+b` is :) I just found out in spec, but I will forget soon.

**Comment by evilebottnawi** [comment]
@hudochenkov we can clarify this in rule description :smile:

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to create the rule [as a plugin](https://stylelint.io/developer-guide/plugins).
